pyproject.toml

```diff
@@ -48,7 +48,8 @@ dependencies = [
     "pyts",
     "torch",
     "torchdiffeq",
-    "toml"
+    "toml", 
+    "lightning"
```

Comment:
Let's make lightning an optional dependency under `[project.optional-dependencies]` as `lightning = ["lightning"]`

Reply:
we might need lightning as core dependency because it is imported in constraints.py